### ISSUE TITLE
ImportError: cannot import name return_values

### ISSUE THREAD

**Issue by sinu** [networktocode/fortimanager-ansible#68]
Team ,

any idea about the below error

```

he full traceback is:
Traceback (most recent call last):
  File "/root/.ansible/tmp/ansible-tmp-1558110708.74-201937228005129/AnsiballZ_fortimgr_address_map.py", line 114, in <module>
    _ansiballz_main()
  File "/root/.ansible/tmp/ansible-tmp-1558110708.74-201937228005129/AnsiballZ_fortimgr_address_map.py", line 106, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File "/root/.ansible/tmp/ansible-tmp-1558110708.74-201937228005129/AnsiballZ_fortimgr_address_map.py", line 49, in invoke_module
    imp.load_module('__main__', mod, module, MOD_DESC)
  File "/tmp/ansible_fortimgr_address_map_payload_aJnZJP/__main__.py", line 296, in <module>
ImportError: cannot import name return_values

failed: [] (item={u'subnet': u'10.10.100.10/32', u'type': u'subnet', u'name': u'db01', u'vdom': u'prod', u'fg': u'prod_dmz'}) => {
    "ansible_loop_var": "item",
    "changed": false,
    "item": {
        "fg": "prod_dmz",
        "name": "db01",
        "subnet": "10.10.100.10/32",
        "type": "subnet",
        "vdom": "prod"
    },
    "module_stderr": "Traceback (most recent call last):\n  File \"/root/.ansible/tmp/ansible-tmp-1558110708.74-201937228005129/AnsiballZ_fortimgr_address_map.py\", line 114, in <module>\n    _ansiballz_main()\n  File \"/root/.ansible/tmp/ansible-tmp-1558110708.74-201937228005129/AnsiballZ_fortimgr_address_map.py\", line 106, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/root/.ansible/tmp/ansible-tmp-1558110708.74-201937228005129/AnsiballZ_fortimgr_address_map.py\", line 49, in invoke_module\n    imp.load_module('__main__', mod, module, MOD_DESC)\n  File \"/tmp/ansible_fortimgr_address_map_payload_aJnZJP/__main__.py\", line 296, in <module>\nImportError: cannot import name return_values\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
    "rc": 1
}
fatal: []: FAILED! => {
    "msg": "The task includes an option with an undefined variable. The error was: 'dict object' has no attribute 'subnet'\n\nThe error appears to be in '/root/fortigate/fortimanager-ansible-master/test.yml': line 8, column 7, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n  tasks:\n    - name: CONFIGURE ADDRESS MAPPINGS\n      ^ here\n"
}

```

**Comment by jmcgill298** [comment]
can you share your playbook?

**Comment by sinu** [comment]
> 
> 
> can you share your playbook?

@jmcgill298 this is nothing about the playbook this about the basic module which you are invoking. the latest version ansible 2.8 removed definitions for return_values from basics.py. can we fix this ?
`
ImportError: cannot import name return_values`

**Comment by 514amir** [comment]
We are also getting this issue

- name: "Get facts from Fortimanager"
  fortimgr_facts:
    provider: "{{ va_fortimanager_cfg.credentials }}"
    fortigates: "all"
    config_filter: "all"
  register: fortimanager_facts

**Comment by 514amir** [comment]
We just updated to latest ansible tower on red hat (we migrated from an old instance)

**Comment by asher2199** [comment]
I checked the master branch of Ansible: https://github.com/ansible/ansible

return_values has been removed:
```
commit aba4bed803fc783ea920798356eee196f7b56129
Author: Sam Doran <sdoran@redhat.com>
Date:   Mon Mar 4 17:02:44 2019 -0500

    Move _handle_no_log_values() out of basic.py (#48628)

    * Rename method and make private
    * Use is_iterable, combine transformations
    * Remove unused return_values from network modules
    * Improve docstrings in new functions
    * Add new PASS_VAR
    * Add unit tests for list_no_log_values
    * Fix unit tests for Python 2.6

-def return_values(obj):
-    """ Return native stringified values from datastructures.
-
-    For use with removing sensitive values pre-jsonification."""
-    if isinstance(obj, (text_type, binary_type)):
-        if obj:
-            yield to_native(obj, errors='surrogate_or_strict')
-        return
-    elif isinstance(obj, SEQUENCETYPE):
-        for element in obj:
-            for subelement in return_values(element):
-                yield subelement
-    elif isinstance(obj, Mapping):
-        for element in obj.items():
-            for subelement in return_values(element[1]):
-                yield subelement
-    elif isinstance(obj, (bool, NoneType)):
-        # This must come before int because bools are also ints
-        return
-    elif isinstance(obj, NUMBERTYPES):
-        yield to_native(obj, nonstring='simplerepr')
-    else:
-        raise TypeError('Unknown parameter type: %s, %s' % (type(obj), obj))
```

**Comment by sinu** [comment]
@514amir you can add "def return_values" session to basic.py and issue will be resolved

**Comment by izenoamos** [comment]
Having same issue.

ansible-playbook 2.8.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/usr/share/ansible/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible-playbook
  python version = 2.7.5 (default, Sep 12 2018, 05:31:16) [GCC 4.8.5 20150623 (Red Hat 4.8.5-36)]

**Comment by izenoamos** [comment]
> @514amir you can add "def return_values" session to basic.py and issue will be resolved

thanks for this

**Comment by jmcgill298** [comment]
I got a chance to look into this today, and started working on a fix in #72 I still need to update several modules and test the updates. I tested with fortimgr_facts, and it seemed to work fine in Ansible2.4 and above.

**Comment by jmcgill298** [comment]
#72
